packages/next/src/server/base-server.ts

```diff
@@ -911,7 +911,8 @@ export default abstract class Server<ServerOptions extends Options = Options> {
         )
       }
 
-      req.headers['x-forwarded-host'] ??= `${this.hostname}:${this.port}`
+      req.headers['x-forwarded-host'] ??=
+        req.headers['host'] ?? `${this.hostname}:${this.port}`
```

Comment:
Did a similar fix in #58399 - should this be this.hostname rather than having this.port too?

Regardless - this fix solves the same proble

Reply:
@eyn I feel there's a world where `${this.hostname}:${this.port}` makes sense, however, I think that you're correct in that it should be `this.hostname`. So as to revert back to the expected functionality and not cause potential downstream issues. The port is still available within other headers.

Ref: [RFC-9110 Section 7.2](https://datatracker.ietf.org/doc/html/rfc9110#section-7.2)